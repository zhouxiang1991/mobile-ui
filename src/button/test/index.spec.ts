import { mount } from '@vue/test-utils'
import { Button } from '..'

describe('Button', () => {
  test('should emit event', async () => {
    const wrapper = mount(Button, {})

    await wrapper.find('button').trigger('click')
    expect(wrapper.emitted('click')).toHaveLength(1)
  })
  test('should contains some text', () => {
    const wrapper = mount(Button, {
      slots: {
        default: 'my-button',
      },
    })

    expect(wrapper.find('button').text()).toBe('my-button')
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('should contains some className', () => {
    const wrapper = mount(Button, {
      props: {
        size: 'small',
        type: 'primary',
      },
    })

    expect(wrapper.find('button').classes()).toContain('zx-button--small')
    expect(wrapper.find('button').classes()).toContain('zx-button--primary')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
