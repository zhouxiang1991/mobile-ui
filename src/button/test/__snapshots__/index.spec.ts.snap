// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should contains some className 1`] = `
"<button class=\\"zx-button--primary zx-button--small\\">默认按钮</button>
<!-- <Common></Common> -->
<!-- <img :src=\\"Echarts\\"> -->"
`;

exports[`Button should contains some text 1`] = `
"<button class=\\"zx-button--default zx-button--normal\\">my-button</button>
<!-- <Common></Common> -->
<!-- <img :src=\\"Echarts\\"> -->"
`;

exports[`Button should match original snapshoot 1`] = `
"<button class=\\"zx-button--default zx-button--normal\\">默认按钮</button>
<!-- <Common></Common> -->
<!-- <img :src=\\"Echarts\\"> -->"
`;
